src/nupic/algorithms/SpatialPooler.cpp

```diff
@@ -321,7 +321,9 @@ void SpatialPooler::setPotential(UInt column, const UInt potential[]) {
 
 void SpatialPooler::getPermanence(UInt column, Real permanences[]) const {
   NTA_ASSERT(column < numColumns_);
-  permanences_.getRowToDense(column, permanences);
+  // TODO: Check that getRowToDense_itr() is the right function to call.
+  //       The function is expecting an iterator. This could be correct.
```

Comment:
verify and rm comment

Reply:
> verify and rm comment  

This was a little strange.  getRowToDense() is overloaded and accepts an iterator or a vector (See py_SparseMatrix.cpp line 181).  In this place it is being called with an array pointer.  The compiler resolves the pointer to the iterator.  I was not sure if that was really the intent.

pyBind11 requires all instances of overloading be declared with their signatures so it can tell them apart.

I was renaming the iterator signature of getRowToDense() because I did not want it exposed to Python.  I could be wrong on this.  Maybe the iterator IS the one I want to expose to Python and accept just a pointer....in which case it should have an pointer signature. I was hoping that you two would have better insight into what the code is trying to do here.  There are several overloaded calls like this. (See py_SparseMatrix.cpp line 264).

Reply:
I'll have a look, but my notion is: 
- if written properly, both/any-of the methods should work
- we hope to remove SparseMatrix in the future, so it shouldn't matter anyway